src/Swarm/Game/Step.hs

```diff
@@ -111,12 +111,14 @@ gameTick = do
             when (newLoc /= oldLoc) $ do
               robotsByLocation . at oldLoc %= deleteOne rn
               robotsByLocation . at newLoc . non Empty %= IS.insert rn
+            time <- use ticks
             case waitingUntil curRobot' of
-              Just wakeUpTime ->
-                sleepUntil rn wakeUpTime
+              Just wakeUpTime
+                -- if w=2 t=1 then we do not needlessly put robot to waiting queue
+                | wakeUpTime - 2 <= time -> return ()
```

Comment:
It doesn't really matter much, but I am curious why you chose `2` here instead of `1`.

Reply:
Because I am very bad with off-by-one errors and 2 seemed safer. It still took me several tries to get it right.

Reply:
That's totally fair. 😄